editor/find_in_files_scanner.cpp

```diff
@@ -0,0 +1,352 @@
+
+		// Let's handle some edge cases!
+		if (matches.size() > matches_offset) {
+			// We have new matches
+
+			// Check if the first new match is on the first line, if so handle it separately
+			if (matches.get(matches_offset).line_number == 1) {
+				FindInFilesScanner::ScanMatch scan_match = matches.get(matches_offset);
+				if (scan_match.begin - SCRIPT_SOURCE_TAG_LEN == scan_match.end - SCRIPT_SOURCE_TAG_LEN - 1) {
+					// Edge case matching the double quote mark
+					matches.remove_at(matches_offset);
+				} else {
+					// Another edge case where we need to remove "script/source = " from the resulting line
+					String actual_line = scan_match.line.replace("script/source = \"", "");
+					scan_match.file_path = relative_file_path;
+					scan_match.begin = scan_match.begin - SCRIPT_SOURCE_TAG_LEN;
+					scan_match.end = scan_match.end - SCRIPT_SOURCE_TAG_LEN;
+					scan_match.line = actual_line;
+					matches.insert(matches_offset, scan_match);
+					matches.remove_at(matches_offset + 1);
+				}
+			}
+
+			// Unescape all matched lines, this is necessary because of the way builtin scripts are stored.
+			// They get escaped before being written into the file, so things like quotes gets encoded into \".
+			int i = 0;
+			for (auto &scan_match : matches.slice(matches_offset)) {
+				String unescaped_line = scan_match.line.c_unescape();
+				int diffs_found = 0;
+				for (int j = 0; j <= scan_match.end; ++j) {
+					if (scan_match.line[j] == unescaped_line[j - diffs_found]) {
+						continue;
+					}
+
+					diffs_found++;
+				}
+				scan_match.begin -= diffs_found;
+				scan_match.end -= diffs_found;
+				scan_match.line = unescaped_line;
+				matches.insert(matches_offset + i, scan_match);
+				matches.remove_at(matches_offset + i + 1);
+				i++;
+			}
+
+			matches_offset = matches.size();
+		}
+	}
+}
+
+void BuiltinScriptFindInFilesScanner::replace(Ref<FileAccess> file, String file_path, Vector<FindInFilesScanner::ScanLocation> &locations, bool match_case, bool whole_words, String search_text, String new_text) {
+	for (auto &match : _parse_tscn(file)) {
+		if ((file->get_path() + "::" + match.key) != file_path) {
+			continue;
+		}
+
+		file->seek(0);
+		// Account for the edge cases also found during scanning
+		for (auto &location : locations) {
+			if (location.line_number == 1) {
+				location.begin += SCRIPT_SOURCE_TAG_LEN;
+				location.end += SCRIPT_SOURCE_TAG_LEN;
+			}
+		}
+		DefaultFindInFilesScanner::replace(file, file->get_path(), locations, match_case, whole_words, search_text, new_text, match.value.script_start_line - 1);
+	}
+}
+
+HashMap<String, BuiltinScriptFindInFilesScanner::SubResource> BuiltinScriptFindInFilesScanner::_parse_tscn(Ref<FileAccess> file) {
+	enum BuiltinScriptMarkers {
+		SUB_RESOURCE_MARKER,
+		SCRIPT_SOURCE_MARKER,
+		SCRIPT_EOF_MARKER
+	};
+	HashMap<BuiltinScriptMarkers, String> builtin_script_markers_map;
```

Comment:
I think it would be better if this map was built only once, even as a local static variable.